src/audio/dai-zephyr.c

```diff
@@ -1055,6 +1055,10 @@ static int dai_comp_trigger_internal(struct dai_data *dd, struct comp_dev *dev,
 
 	comp_dbg(dev, "dai_comp_trigger_internal(), command = %u", cmd);
 
+	ret = comp_set_state(dev, cmd);
+	if (ret < 0)
+		return ret;
+
```

Comment:
first of all this affects every used of this function, right? From the commit description it sounds like everything should be broken ATM because `dai_copy()` is never called, which probably isn't the case. @singalsu do you know off the top of your head how this is working currently on Intel systems? I'd need to do some debugging to check that.
Secondly, the rest of the function can still fail, in which case this state change isn't reverted. *If* this indeed ends up being done this way, you probably need to add error handling and return the state to the previous one

Reply:
> first of all this affects every used of this function, right? From the commit description it sounds like everything should be broken ATM because dai_copy() is never called, which probably isn't the case. @singalsu do you know off the top of your head how this is working currently on Intel systems? I'd need to do some debugging to check that.

yeah but doesn't Intel use the copier component or a combination of the 2? Seems this way from 6ffb0526910e0d3410f3a9dd5b68457d359cc8f4.

> Secondly, the rest of the function can still fail, in which case this state change isn't reverted. _If_ this indeed ends up being done this way, you probably need to add error handling and return the state to the previous one

ACK

Reply:
To be on the safe side, please let @lyakh resolve his own comments:
https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#workflow-suggestions-that-help-reviewers